### PR TITLE
[MM-45666] - A/B Test: Nav Bar Upgrade button

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
This adds a feature flag, `FeatureFlagServerUpgradeButtonText` to control the text to be shown on Plan Upgrade Button for cloud workspaces

FeatureFlagServerUpgradeButtonText = A , the button reads "Upgrade"
FeatureFlagServerUpgradeButtonText = B, the button reads "View plans"

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-45666

#### Related Pull Requests

- Has server changes (https://github.com/mattermost/mattermost-server/pull/20676)

#### Screenshots
Screenshot 2022-07-19 at 12 15 53
Screenshot 2022-07-19 at 12 17 56

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: components/global_header/right_controls/plan_upgrade_button/index.tsx
Comment (neallred) [inline components/global_header/right_controls/plan_upgrade_button/index.tsx:41]: I think the way opening the pricing modal works for playbooks and boards is they import `openPricingModal`. This  means that those elements won't get the variation. Should they? i.e. should we determine what the A/B test value is and assign to `openPricingModal` the result of that?

File: components/common/hooks/useOpenPricingModal.ts
Comment (neallred) [inline components/common/hooks/useOpenPricingModal.ts:18]: Maybe we can call the property `variant`? The value passed in is a name for a variant, and variant is the standard name for an A/B test option (from https://en.wikipedia.org/wiki/A/B_testing: "A/B tests consist of a randomized experiment with two variants")

**Review by neallred — APPROVED** [review]
LGTM pending changing the property name to something like `variant`

File: components/common/hooks/useOpenPricingModal.ts
Comment (marianunez) [inline components/common/hooks/useOpenPricingModal.ts:18]: Do we need to track the feature flag variant here? On the data side, we know what variants a server has from the rest of the config so this can be mapped when analizing the test results

File: components/common/hooks/useOpenPricingModal.ts
Comment (marianunez) [inline components/common/hooks/useOpenPricingModal.ts:17]: We still want to track the pricing modal open event for self-managed

File: components/global_header/right_controls/plan_upgrade_button/index.tsx
Comment (marianunez) [reply components/global_header/right_controls/plan_upgrade_button/index.tsx:41]: I'm wondering if we need the variant on the track event at all - see my comment above

**Review by marianunez — APPROVED** [review]
Thanks @AGMETEOR!

**Comment by marianunez** [comment]
@AGMETEOR once this is merged, let's cherry pick to `cloud` so we can launch the test on the next cloud release

**Comment by marianunez** [comment]
@stevemudie can we prioritize this PR for today? Would like to get this fix in for the upcoming cloud release

**Review by stevemudie — APPROVED** [review]
We can't get this to work properly on the Spinwick so I'll test it post merge.

**Pull request merged**
